yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -60,6 +62,9 @@ private[yarn] class YarnAllocator(
 
   import YarnAllocator._
 
+  // RackResolver logs an INFO message whenever it resolves a rack, which is way too often.
+  Logger.getLogger(classOf[RackResolver]).setLevel(Level.WARN)
```

Comment:
The explanation is a little involved, so will explain here:
The issue with the message in log4j-defaults.properties was that the log4j.properties packaged inside Hadoop was getting picked up instead.  While it would be ideal to fix that as well, we still want to quiet this in situations where a user supplies their own custom log4j properties.